docs/reference/online-stores/elasticsearch.md

```diff
@@ -0,0 +1,120 @@
+# Elasticsearch online store
+
+## Description
+
+The [Elasticsearch](https://www.elastic.co/) online store provides support for materializing feature values into an Elasticsearch database for serving online features.
+
+Features:
+* Creating indexes and documents through parameters in FeatureView
+* Support for float and binary vector embeddings
+* Full support for HNSW indexing for available metric types in Elasticsearch
+
+## Getting started
+In order to use this online store, you'll need to install the elasticsearch extra:
+- `pip install 'feast[elasticsearch]'`
+
+## Example
+
+Connecting to an Elasticsearch instance:
+
+{% code title="feature_store.yaml" %}
+```yaml
+project: my_feature_repo
+registry: data/registry.db
+provider: local
+online_store:
+  type: elasticsearch
+  endpoint: ELASTICSEARCH_URL
+  username: ELASTICSEARCH_USERNAME
+  password: ELASTICSEARCH_PASSWORD
+```
+{% endcode %}
+
+The full set of configuration options is available in [ElasticsearchOnlineStoreConfig](https://github.com/ExpediaGroup/feast/blob/master/sdk/python/feast/expediagroup/vectordb/elasticsearch_online_store.py#L40).
+
+## Elasticsearch Online Store Format
+
+### Overview
+
+Elasticsearch is a document search engine with vector embedding indexing and knn support.
+
+Elasticsearch organizes data into _indexes_. An index contains a series of object documents. A document can contain one vector field one the top-level of a document. A vector can be either an array of floats or binary (bytes). The vector is indexed using Hierarchical Navigable Small World (HNSW) graph.
+
+### Managing Elasticsearch Resources Through Feature View
+
+An index will reflect the schema of a FeatureView. The name of the FeatureView will also be the name of the index in Elasticsearch. The online store implementation translates the schema of the _feature view_ into the equivalent schema of a document.
+
+Vector indexing is defined through the _field_ level tags. The following tags are expected:
+
+* index_type: must be `hnsw`. This tag is used to identify which field should be indexed as a vector embedding.
```

Comment:
how can we support `flat` indexing?

Reply:
Elasticsearch currently only supports HNSW, there is no option for flat.

Reply:
the documentation isn't really clear about it, but the brute force option mentioned [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/knn-search.html#knn-methods) sounds very much like it would be a flat index. It would be used if index is set to `false`: https://www.elastic.co/guide/en/elasticsearch/reference/current/knn-search.html#exact-knn. At least we should offer this as an option. Basically, calling it `index_type` -> `flat` and then set `index` to `false`

Reply:
Ok, I did not see that before. I'll need to make a small code change.